database/db.go

```diff
@@ -763,6 +763,29 @@ func (db *DB) Delete(
 	return db.DeleteStreamed(ctx, entityType, idsCh, onSuccess...)
 }
 
+// RunInTx allows running a function in a database transaction without requiring manual transaction handling.
+//
+// A new transaction is started on [DB] which is then passed to fn. After fn returns, the transaction is
+// committed unless an error was returned. If fn returns an error, that error is returned or when failing
+// to start or/and commit the transaction.
+func (db *DB) RunInTx(ctx context.Context, fn func(context.Context, *sqlx.Tx) error) error {
```

Comment:
I would rename that function to `ExecTx`. Isolation level should also be adjustable. Please also consider that we already have special handling for serializable transactions.

Reply:
> Please also consider that we already have special handling for serializable transactions.

Which special handling are you referring to? As far as I am aware, our HA tx utilises different method for MySQL driver to simulate a serializable transaction. However, I don't see how this method should make any indication of that.

Reply:
> However, I don't see how this method should make any indication of that.

Why not? I don't understand your answer to be honest.

Reply:
This method only executes the provided callback and never gets to see what queries you are executing. So how should I append `LOCK IN SHARE MODE` to every single query that is executed within that callback?

Reply:
Therefore, I would just allow to pass `sql.TxOptions` as an argument to this method, and the caller has to perform the special cases required for that callback.

Reply:
We'll introduce another version of `ExecTx` in the future that takes care of the tx isolation special handling, but this method won't change as it only invokes the provided callback with a newly initialised tx and has no knowledge of the actual queries executed by `fn`.